--- MODULE.bazel
+++ MODULE.bazel
@@ -4,7 +4,7 @@
     compatibility_level = 1,
 )
 
-bazel_dep(name = "rules_python", version = "0.30.0")
+bazel_dep(name = "rules_python", version = "0.24.0")
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
 python.toolchain(
@@ -17,6 +17,7 @@
 )
 
 pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
+
 pip.whl_mods(
     # we are using the appended_build_content.BUILD file
     # to add content to the request wheel BUILD file.
