--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,16 @@
+
+cc_library(
+    name = "implot",
+    srcs = glob(["*.cpp"]),
+    hdrs = glob(["*.h"]),
+    deps = [
+        "@imgui"
+    ],
+    strip_include_prefix = ".",
+    visibility = ["//visibility:public"],
+    copts = select({
+		"@bazel_tools//src/conditions:darwin": [],
+        "@bazel_tools//src/conditions:windows": ["/wd4305"],
+        "@rules_bzlmodrio_toolchains//constraints/combined:is_linux": ["-Wno-format-nonliteral"],
+    }),
+)
