--- MODULE.bazel
+++ MODULE.bazel
@@ -8,9 +8,9 @@
 bazel_dep(name = "rules_cc", version = "0.0.9")
 bazel_dep(name = "rules_java", version = "7.6.5")
 bazel_dep(name = "rules_bzlmodrio_toolchains", version = "2024-1.bcr1")
-bazel_dep(name = "bzlmodrio-allwpilib", version = "2025.1.1-beta-1")
+bazel_dep(name = "bzlmodrio-allwpilib", version = "2024.1.1")
 bazel_dep(name = "bzlmodrio-opencv", version = "2024.4.8.0-4.bcr1")
-bazel_dep(name = "bzlmodrio-ni", version = "2025.0.0")
+bazel_dep(name = "bzlmodrio-ni", version = "2024.2.1.bcr1")
 
 setup_bzlmodrio_navx_cpp_dependencies = use_extension("//:maven_cpp_deps.bzl", "setup_bzlmodrio_navx_cpp_dependencies")
 use_repo(
